### PR TITLE
chore: exclude pandas 2.2.x from testing

### PR DESCRIPTION
**Pull request by tswast**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Towards internal issue 322845272 🦕

### CODE REVIEW

**Review by chelsea-lin — APPROVED** [review]
LTGM

File: noxfile.py [hunk 1]
```diff
@@ -551,7 +551,7 @@ def prerelease(session: nox.sessions.Session, tests_path):
         # https://github.com/googleapis/python-bigquery-dataframes/issues/341
         # https://github.com/googleapis/python-bigquery-dataframes/issues/337
         # are resolved
-        "pandas!=2.1.4, !=2.2.0rc0, !=2.2.0",
+        "pandas<2.1.4",
```
Comment (shobsi) [inline noxfile.py:554]: JFYI we have been following `!=` approach to track the exact breaking changes with each release via the associated follow up items (in the comments above) based on the following feedback.

https://github.com/googleapis/python-bigquery-dataframes/pull/268#discussion_r1423205172

**Review by shobsi — APPROVED** [review]
If we have narrowed down a new breaking issue in 2.2.1, worth adding in the comments like other excluded versions

**Comment by tswast** [comment]
> If we have narrowed down a new breaking issue in 2.2.1, worth adding in the comments like other excluded versions

I don't think it's anything new compared to 2.2.0 as far as I can tell.

**Pull request merged**